main.py

```diff
@@ -246,28 +260,50 @@ def animate_df(df, title, ylabel, xlabel, has_custom_color, color_hex):
     x_min = min_bar - (max_bar - min_bar) * 0.01
 
     fig, ax = plt.subplots()
+  
+  num_bars = len(df.columns)
+  colors   = rand_colors(num_bars, min_val=0.5,    max_val=0.9)
+   
+  fig = plt.figure()
+
+  def draw_graph(frame):
+    plt.cla()
+    ax = plt.axes(label=str(frame))
+    length = len(df)
+    my_list =[]
+    categories = []
+    new_colors = []
+
+    series        = df.iloc[frame] #selects ith row
+    rank          = series.rank(method = 'first', ascending=0)
+
+    for i in range(1,bars_shown+1):
+      for j in range(len(series)):
+        if rank[j] == i:
+          my_list.append(series[j])
+          categories.append(series.index[j])
+          new_colors.append(colors[j])
+    
+    values = my_list # x-axis
+   
+    max_bar = df.max().max()
+    min_bar = df.min().min()
 
-    def draw_graph(frame):
-      # clear the current axes
-      plt.cla()
+    x_max = max_bar + (max_bar - min_bar) * 0.05
+    x_min = min_bar - (max_bar - min_bar) * 0.01
 
-      # selects ith row
-      series = df.iloc[frame]
-      rank = series.rank(method='first')
-      # to do: only show certain max number of bars
-      categories = series.index
-      # y-axis
-      values = series.array
+       
+    ax.set_xlim(left=x_min, right=x_max)
+    ax.barh(range(bars_shown+1,1,-1), values, tick_label=categories, color=new_colors)
 
-      ax.set_xlim(left=x_min, right=x_max)
-      if has_custom_color:
-        ax.barh(rank, values, tick_label=categories, color=color_hex)
-      else:
-        ax.barh(rank, values, tick_label=categories, color=colors)
+    if has_custom_color:
+      ax.barh(rank, values, tick_label=categories, color=color_hex)
+     else:
+      ax.barh(rank, values, tick_label=categories, color=colors)
```

Comment:
Combine this section with the above line to incorporate both bar number and custom color:
```suggestion
    if has_custom_color:
      ax.barh(range(bars_shown+1,1,-1), values, tick_label=categories, color=color_hex)
     else:
      ax.barh(range(bars_shown+1,1,-1), values, tick_label=categories, color=new_colors)
```